src/Server/Resource/DoctrineResource.php

```diff
@@ -319,9 +320,10 @@ public function getHydrator()
      * Create a resource
      *
      * @param mixed $data
+     * @param EntityFactoryInterface $entityFactory Optional factory to handle instantiation of the entity
      * @return ApiProblem|mixed
      */
-    public function create($data)
+    public function create($data, EntityFactoryInterface $entityFactory = null)
```

Comment:
`DoctrineResource` is already acting like a factory (`new $entityClass`), which is one responsibility too many IMO. This PR actually segregates that outside of the responsibilities of this class.